Documentation/41132 add guide on how to setup openproject bim on docker

adam-op opened the pull request:
[#41132] Add guide on how to setup OpenProject BIM on Docker

https://community.openproject.org/work_packages/41132

Commit:
[#41132] Add guide on how to setup OpenProject BIM on Docker

[#41132] Add guide on how to setup OpenProject BIM on Docker

https://community.openproject.org/work_packages/41132

Commit:
Update README.md

Commit:
reorganization of the location for these BIM docker docs

[#41132] Add guide on how to setup OpenProject BIM on Docker

https://community.openproject.org/work_packages/41132

Commit:
[#41132] Add guide on how to setup OpenProject BIM on Docker

[#41132] Add guide on how to setup OpenProject BIM on Docker

https://community.openproject.org/work_packages/41132

wielinde commented on docs/installation-and-operations/installation/docker/README.md:
```diff
@@ -70,6 +70,16 @@ your `docker-compose.yml` file is stored (`compose` in this case).
+
+In the compose folder you will find the file `docker-compose.yml` which can be edited. Please be aware that only those variables shall be edited which are documented as not everything is meant to be configured or bend.
+
+##### BIM Edition
```
That title has 5 hashes, while its parent has 3 hashes. Is there something I don't understand or is there one level missing?
```suggestion
#### BIM Edition
```

adam-op replied:
done

wielinde commented on docs/installation-and-operations/installation/docker/README.md:
```diff
@@ -70,6 +70,16 @@ your `docker-compose.yml` file is stored (`compose` in this case).
 
+### Configuration
+
+In the compose folder you will find the file `docker-compose.yml` which can be edited. Please be aware that only those variables shall be edited which are documented as not everything is meant to be configured or bend.
```
@machisuji is that actually the way to go, or would you recommend using a `docker-compose.override.yml` for configuration?

adam-op replied:
waiting for @machisuji

machisuji replied:
@adam-op It's best to use a `docker-compose.override.yml` as @wielinde suggested so users can just download the latest `docker-compose.yml` from the deploy repo without having to worry about losing their own changes.

adam-op replied:
okay, I will rewrite the docs with using the mentioned **docker-compose.override.yml**

adam-op replied:
I did the requested changes, please could you approve/merge? Thank you

wielinde commented on docs/installation-and-operations/changing-to-bim-edition/README.md:
````diff
@@ -77,3 +77,33 @@ OpenProject installations with a valid Enterprise Edition token can switch to th
+    IMAP_ENABLED: "${IMAP_ENABLED:-false}"
+```
+
+Note: If your current Docker installation does not yet hold important information I recommend you to simply create all docker containers from scratch as the seeded data such as themes, types, and demo projects are different in the BIM edition. The seed data gets seeded only the very first time you call.
````
Do we write from the the first person's perspective? Looks weard to me to see "I recommend" in a guide.

adam-op replied:
Hi @wielinde 
actually i might have found an answer to your question.
There is a Style Guide that I will need to read, no one requested that yet, so i did not know:
Thanks for asking:
https://www.openproject.org/docs/development/contribution-documentation/documentation-style-guide/

machisuji replied:
@adam-op I would also prefer another way to say this. I don't think the style guide says anything about this yet but still.

Suggestion: "[...] important information, it would be easiest to re-create all containers from scratch"

You may also need to clarify that starting from scratch will also require to actually delete the volumes (`docker-compose down --volumes`).

adam-op replied:
- First person will be replaced by e.g. usage of passive.
- deleting the volumes will be added as suggested

wielinde commented on docs/installation-and-operations/changing-to-bim-edition/README.md:
```diff
@@ -77,3 +77,33 @@ OpenProject installations with a valid Enterprise Edition token can switch to th
+
+##### OpenProject BIM Edition with Docker Compose
+
+If you would like to use OpenProject BIM Edition inside your docker-compose containers, in your `docker-compose.yml` file in `x-op-app` > `environment` add the line
```
As asked below, wouldn't it be a good idea to use a `docker-compose.override.yml` file for that sort of configuration/adoptions? @machisuji?

adam-op replied:
@machisuji please look into this

wielinde commented on docs/installation-and-operations/changing-to-bim-edition/README.md:
```diff
@@ -77,3 +77,33 @@ OpenProject installations with a valid Enterprise Edition token can switch to th
+
+##### OpenProject BIM Edition with Docker Compose
+
+If you would like to use OpenProject BIM Edition inside your docker-compose containers, in your `docker-compose.yml` file in `x-op-app` > `environment` add the line
```
I have doubts that such a concept of a "docker-compose container" exists. There are docker containers.

adam-op replied:
true its not a container, its an docker-compose installation
If you would like to use BIM Edition inside your docker-compose  OpenProject installation.

adam-op replied:
done

wielinde commented on docs/installation-and-operations/changing-to-bim-edition/README.md:
```diff
@@ -77,3 +77,33 @@ OpenProject installations with a valid Enterprise Edition token can switch to th
+
+### Docker
+
+##### OpenProject BIM Edition with Docker Compose
```
As mentioned below: the markdown headline hierarchy jumps from 3 hashes to 5, missing out 4. That looks semantically wrong to me.

adam-op replied:
Question: who is a normal docker administrator? i suppose we do not have any in OP because they are all involved as developers too

adam-op replied:
done

wielinde reviewed (commented):
@adam-op please find a couple of inline comments regarding the form of the PR. Then I added the question if it would be wise to use `docker-compose.override.yml` files for configurations (I just recently learned about them from @Kharonus). I am not an expert in that topic. Maybe @machisuji has an opinion about it.

Then, we should have a lookout for a normal docker administrator and check that he/she can follow the instructions successfully.

ghost commented:
> Then, we should have a lookout for a normal docker administrator and check that he/she can follow the instructions successfully.

Question: who is a normal docker administrator? i suppose we do not have any in OP because they are all involved as developers too

Commit:
carried out requested changes

[#41132] Add guide on how to setup OpenProject BIM on Docker

https://community.openproject.org/work_packages/41132

Commit:
corrective actions as agreed

corrective actions as agreed

ghost commented:
@machisuji @wielinde  i did the requested changes, please could you approve/merge? Thank you

machisuji reviewed (approved):
👍

machisuji committed:
Merge branch 'release/12.2' into documentation/41132-add-guide-on-how-to-setup-openproject-bim-on-docker
